muikku-core-plugins/src/main/java/fi/otavanopisto/muikku/plugins/notes/NotesRESTService.java

```diff
@@ -130,26 +153,59 @@ private NoteRestModel toRestModel(Note note) {
   @GET
   @Path("/owner/{OWNER}")
   @RESTPermit (handling = Handling.INLINE, requireLoggedIn = true)
-  public Response getNotesByOwner(@PathParam("OWNER") Long owner) {
-    
-    List<Note> notes = notesController.listByOwner(owner);
+  public Response getNotesByOwner(@PathParam("OWNER") Long owner, @QueryParam("listArchived") @DefaultValue ("false") Boolean listArchived) {
+
+    List<Note> notes = notesController.listByOwner(owner, listArchived);
     List<NoteRestModel> notesList = new ArrayList<NoteRestModel>();
+    OffsetDateTime inLastTwoWeeks = OffsetDateTime.now().minusDays(NOTES_FROM_THE_TIME);
     
     for (Note note : notes) {
       NoteRestModel noteRest = toRestModel(note);
       
-      notesList.add(noteRest);
+      UserEntity creator = userEntityController.findUserEntityById(note.getCreator());
+      EnvironmentRoleArchetype creatorUserRole = getUserRoleArchetype(creator.defaultSchoolDataIdentifier());
+      EnvironmentRoleArchetype loggedUserRole = getUserRoleArchetype(sessionController.getLoggedUser());
+      
+      if (loggedUserRole == null || creatorUserRole == null) {
+        return Response.status(Status.BAD_REQUEST).build();
+      }
+      
+      if (Boolean.TRUE.equals(listArchived)) {
+        if (!note.getLastModified().before(Date.from(inLastTwoWeeks.toInstant()))) {
+          if (loggedUserRole.equals(EnvironmentRoleArchetype.STUDENT)) {
+            notesList.add(noteRest);
+          } else {
+            if (!creatorUserRole.equals(EnvironmentRoleArchetype.STUDENT)) {
+              notesList.add(noteRest);
+            }
+          }
+        }
+      } else {
+        if (loggedUserRole.equals(EnvironmentRoleArchetype.STUDENT)) {
+          notesList.add(noteRest);
+        } else {
+          if (!creatorUserRole.equals(EnvironmentRoleArchetype.STUDENT)) {
+            notesList.add(noteRest);
+          } 
+        }
+      }
     }
     
     return Response.ok(notesList).build();
   }
   
-  // mApi() call (mApi().notes.note.del(noteId))
+  private EnvironmentRoleArchetype getUserRoleArchetype(SchoolDataIdentifier userSchoolDataIdentifier) {
+    EnvironmentRoleEntity roleEntity = userSchoolDataIdentifierController.findUserSchoolDataIdentifierRole(userSchoolDataIdentifier);
+    EnvironmentRoleArchetype userRoleArchetype = roleEntity != null ? roleEntity.getArchetype() : null;
+    return userRoleArchetype;
+  }
   
-  @DELETE
-  @Path ("/note/{NOTEID}")
+  // mApi() call (mApi().notes.note.archive.update(noteId))
+  // In this case archiving means moving note to 'trash'. So it's only update method and user can restore the note.
+  @PUT
+  @Path ("/note/{NOTEID}/archive")
```

Comment:
Seemingly this endpoint is used to both archive and unarchive notes? If so, code would be more readable if the path (and method name) was toggleArchived or something.